momento-sdk/src/main/java/momento/sdk/SubscriptionWrapper.java

```diff
@@ -0,0 +1,207 @@
+      // The first message to a new subscription will always be a heartbeat.
+      if (item.getKindCase() != _SubscriptionItem.KindCase.HEARTBEAT) {
+        throw new InternalServerException(
+            "Expected heartbeat message for topic "
+                + topicName
+                + " on cache "
+                + cacheName
+                + ". Got: "
+                + item.getKindCase());
+      }
+      subscribed = true;
+    }
+
+    // Handle different cases based on item
+    switch (item.getKindCase()) {
+      case ITEM:
+        handleSubscriptionItemMessage(item);
+        break;
+      case DISCONTINUITY:
+        handleSubscriptionDiscontinuity(item);
+        break;
+      case HEARTBEAT:
+        handleSubscriptionHeartbeat();
+        break;
+      default:
+        handleSubscriptionUnknown();
+        break;
+    }
+  }
+
+  private void handleSubscriptionDiscontinuity(_SubscriptionItem discontinuityItem) {
+    // Handle subscription discontinuity
+    logger.info(
+        cacheName,
+        topicName,
+        discontinuityItem.getDiscontinuity().getLastTopicSequence(),
+        discontinuityItem.getDiscontinuity().getNewTopicSequence());
+  }
+
+  private void handleSubscriptionHeartbeat() {
+    // Handle subscription heartbeat
+    logger.info("heartbeat " + " " + cacheName + " " + topicName);
+  }
+
+  private void handleSubscriptionUnknown() {
+    // Handle subscription unknown
+    logger.info("unknown " + cacheName + " " + topicName);
+  }
+
+  private void handleSubscriptionItemMessage(_SubscriptionItem item) {
+    // Handle subscription item message
+    _TopicValue topicValue = item.getItem().getValue();
+    TopicMessage message;
+
+    switch (topicValue.getKindCase()) {
+      case TEXT:
+        message =
+            handleSubscriptionTextMessage(topicValue.getText(), item.getItem().getPublisherId());
+        this.options.onItem(message);
+        break;
+      case BINARY:
+        message =
+            handleSubscriptionBinaryMessage(
+                topicValue.getBinary().toByteArray(), item.getItem().getPublisherId());
+        this.options.onItem(message);
+        break;
+      default:
+        handleSubscriptionUnknownMessage();
+        break;
+    }
+  }
+
+  private TopicMessage.Text handleSubscriptionTextMessage(String text, String publisherId) {
+    _TopicValue topicValue = _TopicValue.newBuilder().setText(text).build();
+    // Handle subscription text message
+    return new TopicMessage.Text(topicValue, publisherId.isEmpty() ? null : publisherId);
+  }
+
+  private TopicMessage.Binary handleSubscriptionBinaryMessage(byte[] binary, String publisherId) {
+    _TopicValue topicValue =
+        _TopicValue.newBuilder().setBinary(ByteString.copyFrom(binary)).build();
+    // Handle subscription binary message
+    return new TopicMessage.Binary(topicValue, publisherId.isEmpty() ? null : publisherId);
+  }
+
+  private void handleSubscriptionUnknownMessage() {
```

Comment:
i think this a duplicate of https://github.com/momentohq/client-sdk-java/pull/332/files#diff-5d034b9e3e73ef3355324703eceb4081fa0db81bf278c7ce03e6b323d3a4f0a3R148?